### PR TITLE
Add GitHub Actions workflow

### LINKED ISSUE #182: GitHub Actions
I would like to move this project to GitHub Actions which is more flexible and faster.

Can we do that and did you already signup for the GHA beta?

PS: It might be good to move this repo to an org in the future and invite more maintainers =)

### PR DESCRIPTION
**Pull request by asadm**
Closes #182 

Here is an attempt to add the basic test workflow. 
I have also added the `yarn publish` step but it's commented for now and can be enabled by you when you feel like it.

Also, thank you for making this. I am using patch-package for a [project](https://github.com/remoteinterview/zero) and it's wonderful!

### CODE REVIEW

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    strategy:
+      matrix:
+        os: [ubuntu-18.04]
+        node: [13]
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node }}
+    - run: yarn global add lerna
```
Comment (ds300) [inline .github/workflows/push.yml]: I don't think we need lerna

File: .github/workflows/push.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+on: [push]
+name: Test
+jobs:
+  test:
+    name: Ubuntu on node ${{ matrix.node }}
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        os: [ubuntu-18.04]
+        node: [13]
```
Comment (ds300) [inline .github/workflows/push.yml]: let's add 12 and 10 in there too

File: .github/workflows/push.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+on: [push]
+name: Test
+jobs:
+  test:
+    name: Ubuntu on node ${{ matrix.node }}
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        os: [ubuntu-18.04]
```
Comment (ds300) [inline .github/workflows/push.yml]: let's do `windows-latest`, `ubuntu-latest` and `macos-latest` for extra ✔️

File: .github/workflows/push.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      with:
+        node-version: ${{ matrix.node }}
+    - run: yarn global add lerna
+    - run: yarn install
+    - run: git config --global user.email "patch-package@test.com" && git config --global user.name "patch-package"
+    - run: yarn run test
+# Enable the following when ready to publish automatically.
+#     - run: yarn publish
+#       env:
+#         NODE_AUTH_TOKEN: ${{ secrets.YARN_TOKEN }}
```
Comment (ds300) [inline .github/workflows/push.yml]: you can delete these commented-out lines

File: .github/workflows/push.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
+on: [push]
+name: Test
+jobs:
+  test:
+    name: Ubuntu on node ${{ matrix.node }}
+    runs-on: ubuntu-18.04
```
Comment (ds300) [inline .github/workflows/push.yml]:
```suggestion
    runs-on: ${{ matrix.os }}
```

**Comment by asadm** [comment]
Sure, done!

File: .github/workflows/push.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+on: [push]
```
Comment (DanielRuf) [inline .github/workflows/push.yml]: What about pull requests?

**Pull request merged**